toolchain/check/handle_interface.cpp

```diff
@@ -79,15 +79,14 @@ static auto BuildInterfaceDecl(Context& context,
         // TODO: This should be refactored a little, particularly for
         // prev_import_ir_id. See similar logic for classes and functions, which
         // might also be refactored to merge.
-        CheckIsAllowedRedecl(
+        DiagnoseIfInvalidRedecl(
             context, Lex::TokenKind::Interface, existing_interface.name_id,
             RedeclInfo(interface_info, node_id, is_definition),
             RedeclInfo(existing_interface, existing_interface.latest_decl_id(),
                        existing_interface.is_defined()),
             /*prev_import_ir_id=*/SemIR::ImportIRId::Invalid);
 
         // Can't merge interface definitions due to the generic requirements.
-        // TODO: Should this also be mirrored to classes/functions for generics?
         if (!is_definition || !existing_interface.is_defined()) {
```

Comment:
Should we be using `has_definition_started` for consistency here (and a few lines above) too? I don't think it's possible to define an interface inside its own definition, but maybe that could happen during error recovery somehow?

Reply:
I don't think it makes much difference, but changed -- also, cleaned up some code paths that looked obsolete.